### PR TITLE
docs: Add how to implement a new chain

### PR DESCRIPTION
**Pull request by evaporei**
This document describes how developers can add a new chain to `graph-node`.

It doesn't go over everything in detail, but it's a generalized guide that we can start with and improve on top of.

### CODE REVIEW

**Comment by leoyvens** [comment]
If this were in rustdoc rather than a markdown, then we might be able to require that the examples compile, so they won't become outdated.

**Comment by evaporei** [comment]
@leoyvens but I don't think we want that for those, the examples I've provided are incomplete, the traits require a lot of method implementations.

**Comment by leoyvens** [comment]
Alright, we'll just remember to keep this updated.

**Comment by evaporei** [comment]
You know, actually you just gave me an idea, we could use things like `todo!`s and `unimplemented!` macros in the code. Ok next iteration of this should use this then, great idea!!!

**Comment by leoyvens** [comment]
@evaporei isn't that the perfect use case for a cargo [example](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#examples).

**Review by pienkowb — APPROVED** [review]
Looks good to me 👍

File: docs/implementation/add-chain.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+- https://github.com/graphprotocol/graph-node/pull/2272
+- https://github.com/graphprotocol/graph-node/pull/2292
+- https://github.com/graphprotocol/graph-node/pull/2399
+- https://github.com/graphprotocol/graph-node/pull/2411
+- https://github.com/graphprotocol/graph-node/pull/2453
+- https://github.com/graphprotocol/graph-node/pull/2463
+- https://github.com/graphprotocol/graph-node/pull/2755
+
+All new chains, besides the EVM compatible ones, are integrated using [StreamingFast](https://www.streamingfast.io/)'s [Firehose](https://github.com/streamingfast/firehose). The integration consists of chain specific `protobuf` files with the type definitions.
```
Comment (abourget) [inline docs/implementation/add-chain.md:21]: Maybe link to firehose.streamingfast.io

Comment (evaporei) [reply docs/implementation/add-chain.md]: Fixed!

**Pull request merged**